lms/djangoapps/lti_provider/views.py

```diff
@@ -97,7 +97,7 @@ def lti_launch(request, course_id, usage_id):
     for key in lti_keys:
         if key in params:
             lti_params[lti_keys[key]] = params[key]
-    authenticate_lti_user(request, params['user_id'], lti_consumer, lti_params)
+    authenticate_lti_user(request, params['user_id'], params['roles'], lti_consumer, lti_params)
```

Comment:
Right now roles is a required parameter for the launch and if the role is unrecognized we just allow them to log in as a user. Should that be different?